### ISSUE TITLE
Please add chips

### ISSUE THREAD

**Issue by davidm49090** [twbs/bootstrap#31033]
I know this has been nixed a few times, but the time is right to add chips to the UI toolbox. Would make life a lot easier for forms development.

https://material.io/components/chips

**Comment by mdo** [comment]
Just the HTML and CSS, or JS to go with it for specific implementations?

**Comment by davidm49090** [comment]
Thank you for the fast reply.

HTML and CSS would be great, but as long as it plays nicely with jQuery I would be happy.

An ideal solution would be one that would store an ID, so that you could have a contact chip that displays the person's name while holding their email address.

**Comment by davidm49090** [comment]
You could make this my birthday _and_ Christmas gift this year.

**Comment by leandroruel** [comment]
i believe the "close" button needs JS and looks like he works as a checkbox like buttons with toggle state
![chips](https://user-images.githubusercontent.com/2836002/84841899-17040880-b01a-11ea-8558-c7e71b2d9608.png)

**Comment by klutch-kid** [comment]
Is there anything I can do to help? Or is this all under control? 
Awesome idea!

**Comment by davidm49090** [comment]
> Just the HTML and CSS, or JS to go with it for specific implementations?

I know you have a lot of other components to focus on right now, but I'd love to find this under the Christmas tree.

**Comment by prikeshsavla** [comment]
So
Chip Style
Icon on the front
Text label
optional close button/icon
optional checkbox state?

**Comment by davidm49090** [comment]
> So
> Chip Style
> Icon on the front
> Text label
> optional close button/icon
> optional checkbox state?

Yes! I think that's it.

**Comment by prikeshsavla** [comment]
I think I can work on that.

**Comment by AgentSmith0** [comment]
Any updates to this?

**Comment by prikeshsavla** [comment]
I couldn't do it. Can someone else take it up

**Comment by jovitorrevillas** [comment]
Had a go at making what could be the design document for the chip component.

# Chip
## Design
![image](https://user-images.githubusercontent.com/22047407/186839366-d305941d-d930-4832-8e4d-d3cb2c173ab9.png)

## Properties
#### Color
![image](https://user-images.githubusercontent.com/22047407/186837746-62c90425-6b1e-4d3f-9dc8-553e712453f7.png)
```
<span class="chip bg-primary">Primary</span>
<span class="chip bg-secondary">Secondary</span>
<span class="chip bg-success">Success</span>
<span class="chip bg-danger">Danger</span>
<span class="chip bg-warning text-dark">Warning</span>
<span class="chip bg-info text-dark">Info</span>
<span class="chip bg-light text-dark">Light</span>
<span class="chip bg-dark">Dark</span>
```

#### Variant
![image](https://user-images.githubusercontent.com/22047407/186837917-fe90ae40-530b-4697-9b40-5cf7c77c9990.png)
```
<chip class="chip chip-outline-primary">Primary</chip>
<chip class="chip chip-outline-secondary">Secondary</chip>
<chip class="chip chip-outline-success">Success</chip>
<chip class="chip chip-outline-danger">Danger</chip>
<chip class="chip chip-outline-warning">Warning</chip>
<chip class="chip chip-outline-info">Info</chip>
<chip class="chip chip-outline-light">Light</chip>
<chip class="chip chip-outline-dark">Dark</chip>
```

#### Shape
![image](https://user-images.githubusercontent.com/22047407/186838433-5c6d9fad-d5b9-44f2-a63e-2de8569e8955.png)
![image](https://user-images.githubusercontent.com/22047407/186838457-afab3d61-492d-49d3-aec7-df162cfc0ffa.png)
```
<span class="chip">Primary</span>
<span class="chip chip-rounded">Primary</span>
```

#### Size
![image](https://user-images.githubusercontent.com/22047407/186838902-d25ea266-a2d1-4a61-9ca4-db366d05d33f.png)
![image](https://user-images.githubusercontent.com/22047407/186838928-0c9a9584-1a0a-482e-9d35-499f56dc2c77.png)
![image](https://user-images.githubusercontent.com/22047407/186838957-620617fd-e565-4e30-888f-889c2496b455.png)
```
<span class="chip chip-primary chip-lg">Large chip</span>
<span  class="chip chip-secondary chip-lg">Large chip</span>

<span class="chip chip-primary">Default chip</span>
<span class="chip chip-secondary">Default chip</span>

<span class="chip chip-primary chip-sm">Small chip</span>
<span class="chip chip-secondary chip-sm">Small chip</span>
```

## Usage
#### With icon
![image](https://user-images.githubusercontent.com/22047407/186839855-61481383-6acc-4ec0-9aab-8db6959558ee.png)
```
<span class="chip chip-primary">
 <i class="bi bi-emoji-smile"></i>
 Primary
</span>
```

#### Closable
![image](https://user-images.githubusercontent.com/22047407/186839935-62922e37-ff60-475d-89b5-f9504f26f41b.png)
```
<span class="chip chip-primary">
 Primary
 <i class="bi bi-x-circle"></i>
</span>
```

#### With icon and closable
![image](https://user-images.githubusercontent.com/22047407/186839991-3e2b8b17-18f4-4920-a3b4-370dce8c2cb8.png)
```
<span class="chip chip-primary">
 <i class="bi bi-emoji-smile"></i>
 Primary
 <i class="bi bi-x-circle"></i>
</span>
```

#### Selectable
![image](https://user-images.githubusercontent.com/22047407/186840224-e6767463-9323-40d6-9bf7-4cc145844025.png)
```
<span class="chip chip-primary chip-selected">
 <i class="bi bi-envelope-fill"></i>
 Primary
 <i class="bi bi-x-circle"></i>
</span>

<span class="chip chip-primary">
 <i class="bi bi-envelope-fill"></i>
 Primary
 <i class="bi bi-x-circle"></i>
</span>
```

#### With link
![image](https://user-images.githubusercontent.com/22047407/186840300-731321f0-84b5-40d0-b98a-262b2ffc5682.png)
```
<span class="chip chip-primary">
 <i class="bi bi-envelope-fill"></i>
 Primary
 <a href="/" style="color: inherit;">View Item</a>
 <i class="bi bi-x-circle"></i>
</span>
```

**Comment by Pulkit0729** [comment]
Hey @mdo , i would like like give this a try. Please tell me that this is still needed.

**Comment by davidm49090** [comment]
Love the design work so far... my only comment is that chips should have a container so that they wrap. The container should also have an option for placeholder text. Like if a person is typing, the inputs can be added as chips.

**Comment by techabbayi** [comment]
Hey! 👋 Just checking in — is there still interest in adding “chips” to Bootstrap?

I noticed Bootstrap has badges and nav pills, which cover some similar UI elements. Chips are usually interactive (e.g., used in input fields with a remove button), like in Material UI.

Would this be a completely new component, or an extension of badges? I’d love to help clarify and possibly prototype if this is still relevant!

**Comment by davidm49090** [comment]
Yes! Still hoping to see this. Maybe as a new component. At least for my work, it would be much appreciated.

**Comment by coliff** [comment]
I think this can be closed now as Chips will be in v6. :-)

**Comment by davidm49090** [comment]
What’s the timing to see it in action?

**Comment by mdo** [comment]
This year lol, hopefully only a few months of development and testing. Check `v6-dev` branch or the branch preview: https://v6-dev--twbs-bootstrap.netlify.app/docs/5.3/forms/chip-input/. (Lots of docs subject to change.)

**Comment by septatrix** [comment]
> This year lol, hopefully only a few months of development and testing.

Do you already want feedback?^^

<img width="1080" height="2340" alt="Image" src="https://github.com/user-attachments/assets/894cd574-7c3c-4b50-a61e-529ac2960e49" />
